rust/lance-core/src/utils/mask.rs

```diff
@@ -297,6 +297,31 @@ impl DeepSizeOf for RowIdSelection {
     }
 }
 
+impl RowIdSelection {
+    fn union_all(selections: &[&Self]) -> Self {
+        let mut is_full = false;
+
+        let res = Self::Partial(
+            selections
+                .iter()
+                .filter_map(|selection| match selection {
+                    Self::Full => {
+                        is_full = true;
+                        None
+                    }
+                    Self::Partial(bitmap) => Some(bitmap),
+                })
+                .union(),
```

Comment:
this is essentially betting that full is rare. We might want more comprehensive benchmarks to tune the behavior here

Reply:
Yeah `Full` is mostly to represent the deletion prefilter. For bitmap I think your right it should be rare.